listeners.py

```diff
@@ -213,11 +211,12 @@ def __init__(self, get_logfile):
         self.get_logfile = get_logfile
 
     def process_result(self, obj):
-        if not isinstance(obj, WorkerTaskResult):
-            return
+        if isinstance(obj, WorkerTaskResult):
+            if obj.short_status == 'fail' and \
+                    obj.worker_name not in self.failed_workers:
+                self.failed_workers.append(obj.worker_name)
 
-        if obj.short_status in ('fail', 'flaky') and \
-                obj.worker_name not in self.failed_workers:
+        if isinstance(obj, WorkerFlakedTask):
```

Comment:
`and obj.worker_name not in self.failed_workers`?

Reply:
Looks like you are right. Fixed.